sdk/interface.go

```diff
@@ -33,18 +33,6 @@ type SharedParamsClient interface {
 	GetParams(ctx context.Context) (params *sharedtypes.Params, err error)
 }
 
-// BlockClient is the interface to interact with the on-chain block module.
```

Comment:
I think it's weird that we had a nice `interface.go` file for all interface and are now embedding it in `sdk.go`.

I see it either as:
1. Keep in a shared interface file
2. Move into it's own dedicated file alongside the implementation.